projects/s-rxjs-utils/src/lib/map-and-cache-elements.ts

```diff
@@ -4,11 +4,13 @@ import { map } from "rxjs/operators";
 /**
  * Applies a given function to each item in the upstream array and emits the result. Each item is then cached using the key generated by `buildCacheKey` so that the next emission contains references to the matching objects from previous emission, without running `buildDownstreamItem` again. The cache is only held between successive emissions.
  *
+ * If multiple items in your array have the same cache key, it will only call `buildDownstreamItem` once and will return the same object reference for all items.
+ *
  * @param buildCacheKey A function that converts an upstream object into a string to use as the cache key. Needs to return a unique key for each item in the source array.
- * @param buildDownstreamItem A function that converts an upstream object into a downstream object
+ * @param buildDownstreamItem
```

Comment:
`[green]`
You can remove the entire line. TSDoc and editors will still display the type info for it.